### ISSUE TITLE
Documentation: Double hyphens on long options in invocation.html compiled as endashes in HTML builder

### ISSUE THREAD

**Issue by bskinn** [sphinx-doc/sphinx#3781]
~~~*This is a glitch in the documentation output, not in the Sphinx code itself.*~~~

The long options on the ["Invocation" documentation page](http://www.sphinx-doc.org/en/stable/invocation.html) are displayed with only a single hyphen/dash:

```
The sphinx-quickstart script has several options:

-q, –quiet
Quiet mode that will skips interactive wizard to specify options. This option requires -p, -a and -v options.

-h, –help, –version
Display usage summary or Sphinx version.
```
The [`.rst` source](https://raw.githubusercontent.com/sphinx-doc/sphinx/stable/doc/invocation.rst) correctly has two hyphens, but per, e.g., [this Stack Overflow post](http://stackoverflow.com/questions/21574606/how-to-make-appear-in-sphinx-instead-of), it appears that `html_use_smartypants` is at fault.

~~~Since the raw `rst` is not a primary documentation product, inserting a "non-space" between the hyphens may suffice?~~~

~~~`--quiet` --> `-\ -quiet`~~~

This appears to be a general problem in the interaction of `.. option::` and `smartypants`; hacking the arguments to `.. option::` doesn't fix the problem; and in general there may be documentation (e.g., docstrings) where the raw ReST is a first-class documentation object, and a hacky workaround would be undesirable even if it worked.

**Comment by bskinn** [comment]
The following `bash` one-liner prints out a nice summary of all locations in the `.rst` source where a double-hyphen is followed by a letter, which should(?) cover all instances where fixes *might* be needed.

`for fn in $( find . -regex ".*\.rst" ); do grep -ie '--[a-z]' $fn; if [ $? -eq 0 ]; then echo $fn; fi done`

*This comment is obsolete, though possibly still useful.*

**Comment by bskinn** [comment]
@tk0miya After testing, inserting a `\ ` (backslash-space) into the long-option argument **DOES NOT WORK**. Sphinx just renders the two added characters explicitly.  This will probably need to be corrected within the handling of the `.. option::` directive; I searched quite a bit but a good method was not forthcoming.

It could possibly be done by instructing `docutils` not to apply `smartypants` to the arguments of the `.. option::` directive somehow; or perhaps by declaring the arguments as "literal" somehow. I would guess there's something that could be done in the [`Cmdoption(ObjectDescription)`](https://github.com/sphinx-doc/sphinx/blob/d1e8e41b5e4ffd2a4c61dbe8291b18ae22548d25/sphinx/domains/std.py#L147-L202), but I don't know enough of the internals of Sphinx and/or `docutils` to make the needed changes.  Changing [`addnodes.desc_name(optname, optname)`](https://github.com/sphinx-doc/sphinx/blob/d1e8e41b5e4ffd2a4c61dbe8291b18ae22548d25/sphinx/domains/std.py#L169) to `addnodes.literal_strong(optname, optname)` doesn't appear to break anything, but it doesn't intercept the `smartypants` processing of the double-hyphen character sequence. 

~~~Afraid I'm stumped, at least for now.~~~ I think I got it: I added a new node type in `addnodes.py`, `desc_name_literal`, that behaves similarly to `desc_name`, but that also inherits from `docutils.nodes.literal`. I added the necessary `visit_...` and `depart_...` functions to the various writers and changed the relevant node type in `domains/std.py`.  

With the modifications, Sphinx now generates the double-hyphen output correctly in every writer I've tested. The pass/fail performance on the test suite before and after the changes is identical. PR pending my learning `pytest` well enough to write a test for this.

**Comment by jfbu** [comment]
Hi, this is definitely breaking change from 1.5.x.

As a temporary work-around you can use this mark-up:

```
.. option:: -q, -\\-quiet

   Quiet mode that will skips interactive wizard to specify options.
   This option requires `-p`, `-a` and `-v` options.

.. option:: -h, -\\-help, -\\-version

   Display usage summary or Sphinx version.
```

(yes, with two backslashes). Sorry for inconvenience. The work around is for 1.6.1 because on 1.5.6 it generates a warning about duplicate explicit target name.

(edit: there was no issue at 1.5.6; but I said that to point out the workaround I proposed was not a viable solution even for manually maintained documents; anyway the problem will be fixed for 1.6.2 release, with no need to change any markup in sources)


This is caused by Docutils smart quotes processing introduced at #3666.

**Comment by jfbu** [comment]
@bskinn Thanks for the detailed analysis and the researching. @tk0miya is currently fixing this for 1.6.2 release.

**Comment by tk0miya** [comment]
Fixed at #3816  :-)
